### ISSUE TITLE
Email notifications for messages in a room thread don't seem to say who the message is from

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#47081]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.18-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1723051083119829

## Action Performed:
Prerequisite: create room with member
1. Send a message from room
2. Wait for email notification for member regarding the message

## Expected Result:
Message should say from whom it was received
## Actual Result:
Don't seem to say who the message is from
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
![image](https://github.com/user-attachments/assets/5669728c-0b23-4796-8ce8-5c9ec701d95c)
![Snip - #testroom-for-notification-email - mnata utest@gmail com - Gmail - Google Chrome (2)](https://github.com/user-attachments/assets/a67b1130-6a27-42d2-b793-d07312ec0152)

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by anmurali** [comment]
Yea this email notification is super confusing. I think ideally it should look and feel like the product right? Sorta like an IOU or money request related notification does? I received this:
<img width="1067" alt="image" src="https://github.com/user-attachments/assets/a447765a-7d2d-40d6-899c-00d13277a703">


I think it should look like 
<img width="1081" alt="image" src="https://github.com/user-attachments/assets/569fde17-ee5a-4ef9-858f-74f0ae44c9ce">

@Expensify/design what was this designed to look like?

**Comment by shawnborton** [comment]
Hmm I would think we should be showing the sender's avatar + name + full message like we do for other emails. 

Also, in this case, why is the email coming from "Expensify Chat" if that's just the workspace name?

**Comment by anmurali** [comment]
@shawnborton - doesn't my proposal https://github.com/Expensify/App/issues/47081#issuecomment-2287722637 clarify who the sender of each message is? It basically looks like the conversation in that room or group chat itself.

**Comment by shawnborton** [comment]
Hmm I'm not quite following. But I just checked some notifications from #social and does it look like they are behaving as expected? Each new notification comes from an actual person:
![CleanShot 2024-08-28 at 07 26 00@2x](https://github.com/user-attachments/assets/7698dcee-14e6-40f8-b36a-24825dd44100)

I'm not sure how I feel about that tbh. I would think that a group of messages would be sent from Expensify, not a singular person since there are messages from multiple people there. But if just one person sent a message, I can see where it makes sense to have the notification come from just one person?

**Comment by puneetlath** [comment]
It seems to be inconsistent for me. Sometimes it includes the avatars of the senders and sometimes it doesn't. 

Here's an example where notifications for the same room sometimes included the avatar/name and sometimes didn't: 
<img width="905" alt="Screenshot 2024-08-28 at 10 18 04 AM" src="https://github.com/user-attachments/assets/a8820816-f0a7-49bf-970e-7cc95e6481af">

Looks like @Beamanator also experienced similar here: https://expensify.slack.com/archives/C049HHMV9SM/p1724809339227489

**Comment by anmurali** [comment]
Discussing in slack [here](https://expensify.slack.com/archives/C049HHMV9SM/p1725922181804559)

**Comment by jasperhuangg** [comment]
Can look into this, but need to clear off some of the existing PRs that I already have open.

**Comment by jasperhuangg** [comment]
Based on [this](https://github.com/Expensify/Web-Expensify/blob/6b81799d1564f9253a5e89fb33cacdab6e951def/lib/Notification/Report/Comment.php#L359-L380) it looks like we added specific logic to omit the avatars and display names from the notifications if they only contain a single message from a single person. This is consistent with the examples of this issue that we've been seeing in the linked Slack threads.

It looks like this was added [here](https://github.com/Expensify/Web-Expensify/pull/40349) and was probably broken by some other logic that was added afterwards. I'll try to clean up the logic since it's pretty hard to follow atm.

**Comment by puneetlath** [comment]
Oh that's interesting. I feel like it would be better to just have it look consistent every time. Especially now that the messages come from notifications@expensify.com instead of the user themselves. But I'll defer to @shawnborton on that.

**Comment by jasperhuangg** [comment]
> Especially now that the messages come from [notifications@expensify.com](mailto:notifications@expensify.com) instead of the user themselves

Yeah totally agree with this. It seems the reason we added this logic previously was because the sender's name was already in the email's "from" field. Definitely think we should defer to how we're displaying the multi-message notifications for everything now.

**Comment by shawnborton** [comment]
Yeah, I think we originally made 1:1 messages arrive without an avatar and a display name in the body because the idea was that we would send the email as if it came from the sender, and I think David felt quite strongly about making the email experience feel as similar to chat as possible. But some considerations:
- we can't actually show the avatar of the sender
- I think there is a difference that we experience internally when we get these email notifications. For instance, we always have the checkmark and the app icon as our avatar: ![CleanShot 2024-09-12 at 06 47 45@2x](https://github.com/user-attachments/assets/6416f542-e19a-4503-9cfc-929c7489b8c1)

Can someone confirm what it looks like when a non-Expensify account sends a message to someone that generates an email notification? Does that email come from notifications@expensify.com, with an Expensify avatar? If that's the case, I think it makes sense to just completely standardize on the body of the email always containing the user's avatar + display name + message.

**Comment by shawnborton** [comment]
Also cc @Expensify/design for any additional thoughts here - I think we've chatted through this one in the past.

**Comment by shawnborton** [comment]
Okay actually my test just came through - this is what it looks like when my personal account sent a message and generated a notification:
![CleanShot 2024-09-12 at 06 50 10@2x](https://github.com/user-attachments/assets/8e363786-7582-4015-81d4-87732d95c852)

I think the fact that we are always showing the Expensify logo here makes me want to put the user's avatar in the message body, and thus standardize on everything. But notice that we still send the message from my display name. 

I think my vote would be to standardize the message template by always showing an avatar and sender name in the body but we should socialize this idea in Slack first and make sure everyone is on board with it first.

**Comment by dubielzyk-expensify** [comment]
> Can someone confirm what it looks like when a non-Expensify account sends a message to someone that generates an email notification? Does that email come from [notifications@expensify.com](mailto:notifications@expensify.com), with an Expensify avatar? If that's the case, I think it makes sense to just completely standardize on the body of the email always containing the user's avatar + display name + message.

I got this when testing with a friend from Concierge, but this was sent from my friend. Not sure if that's a bug or feature.
![CleanShot 2024-09-12 at 15 23 47@2x](https://github.com/user-attachments/assets/0e1709d3-0d31-4f9c-bd1f-93b04b9f9d22)

My thoughts on this is aligned with you and Danny I believe. I largely find our whole email experience extremely confusing. Some have avatars and some don't, some send from Expensify, some from a user's name.

I think we should be sending the emails all from Expensify with the avatar and sender as `Expensify`, then embed the user avatar and message within the body of the message at the least to make all the messages consistent.

**Comment by dannymcclain** [comment]
I feel like I've talked pretty extensively about this with you two 😅 and I think you both know how dissatisfied I am with our current strategy. TLDR: I'm aligned with you all.

What ever happened to this idea? 👇 

**Email title:**
(E Avatar) Shane Burton via Expensify `<notifications@expensify.com>`

**Email body:**
(Shane's avatar) Shane Burton
Hello! I am not Shawn!

Anyways, I agree with what Jon said here:
>I largely find our whole email experience extremely confusing. Some have avatars and some don't, some send from Expensify, some from a user's name.
>
>I think we should be sending the emails all from Expensify with the avatar and sender as Expensify, then embed the user avatar and message within the body of the message at the least to make all the messages consistent.

**Comment by shawnborton** [comment]
I like that suggestion for the sender Danny, I forgot we had discussed it but yes it does seem quite ideal!

**Comment by jasperhuangg** [comment]
> I think we should be sending the emails all from Expensify with the avatar and sender as Expensify, then embed the user avatar and message within the body of the message at the least to make all the messages consistent.

Strongly agree with what Jon mentioned above and on board with the example that Danny provided above.

> we should socialize this idea in Slack first and make sure everyone is on board with it first.

@shawnborton What do you think is the best way to bring this to Slack, a #whatsnext post? Happy to work on that if needed.

**Comment by jasperhuangg** [comment]
**Problem:**
There is a lot of inconsistency right now with how we format report comment notification emails:
Emails summarizing multiple report comments are sent from “Expensify Chat”/“Expensify Concierge” and include the report commenter’s name/avatar along with the report comment’s text in the email body
<img width="1640" alt="Screenshot 2024-09-13 at 10 22 33 AM" src="https://github.com/user-attachments/assets/e54d9022-2aca-4a5e-b1cf-e9e0ded4bb9b">

Emails summarizing single report comments are sent from the report commenter with the body of the email containing the report comment text
<img width="1636" alt="Screenshot 2024-09-13 at 10 22 51 AM" src="https://github.com/user-attachments/assets/118db1a4-085f-4f6c-a306-146cfd234255">

This leads to a pretty confusing experience that also implies a lack of polish for a couple of reasons:

- It’s strange how some notifications are sent from Expensify Chat, some are sent from Expensify Concierge, and others appear as sent from the report commenter.
- It’s strange how in the second scenario we don’t include the report commenter’s avatar anywhere in the email.

**Solution:**
Standardize all of our report comment notifications to be the first scenario, i.e.
- The email shows as sent from Expensify Chat/Concierge.
- The report commenter’s name and avatar are included with the report comment text in the body of the message.

Some additional notes for completeness:
- Emails should be sent from “Expensify Chat” by default and should only appear as sent from “Expensify Concierge” if the summary is for a Concierge report. It seems we currently use “Expensify Concierge” for report comments sent into any DM which is very confusing [via this](https://github.com/Expensify/Web-Expensify/blob/27afddfc3209674e84333bdaf35080062b53fcfe/lib/ReportUtils.php#L3266).
- The email avatar should always be the Expensify logo, and we’ll include the report commenter’s avatar in the email body.

**Comment by shawnborton** [comment]
Sent this 1:1 but this looks good, but I think my main feedback is:

- When the platform sends you multiple messages that you missed across multiple people in a room or report, the email comes from Expensify
- When the platform sends you a single message or multiple messages that you missed from the same person, the email comes from Member Name via Expensify

**Comment by jasperhuangg** [comment]
Thanks! I went ahead and tested a bunch of scenarios locally and posted my findings in t[his Google Doc](https://docs.google.com/document/d/1QUkuWPqxuGBxdQ2m_qLLHzkljXpIrzfAkfhMfE56VBA/edit).

Also updated the P/S statement based on my findings and sent you the new one on Slack.

**Comment by dannymcclain** [comment]
I think it's also worth noting that AFAIK we only display the proper user avatar in emails for people with an `@expensify.com` email address. Otherwise we always display the Expensify iconmark. Someone can fact check me on that, but I'm pretty sure that's come up a few times. So in your example, where we see Shae's picture as the sender, I believe we're only seeing that because Shae has an `@expensify.com` email address. If it was from `danny@donuts.com`, you'd just see the Expensify logo.

**Comment by jasperhuangg** [comment]
@dannymcclain Yep I confirmed that through my testing, it seems like that's yet another reason why including the avatar/display name in the body of the email would help out a lot with reducing confusion.

**Comment by jasperhuangg** [comment]
I have a P/S statement drafted up with help from @shawnborton that I'm going to send out in the #product room come Monday since it's the weekend right now and no one will see it. Gonna post it here if anyone has any additional feedback before we send it out!

Depending on how that discussion goes it might be worthwhile to do a pre-design to really iron out how we want notifications to look for all the different types of reports and notification scenarios, but we can decide on that later.

**Comment by jasperhuangg** [comment]
Coming from [this CRITICAL issue](https://github.com/Expensify/App/issues/47081) in #newdot-quality

### Problem:
There is a lot of inconsistency/jank with how we format report comment notification emails:
1. Emails summarizing comments from a single user
<img width="989" alt="Screenshot 2024-09-14 at 9 33 24 AM" src="https://github.com/user-attachments/assets/6c01b4f9-3293-4734-b119-2173b93d497f">

- include the report comment text and no information about the commenter in the email body.
- are sent from “<first name>” if the user is an internal employee i.e. from: Jasper
- are sent from “Expensify Chat” if the report is a policy room.
- are sent from “Expensify Concierge” otherwise

2. Emails summarizing comments from multiple users
<img width="1003" alt="Screenshot 2024-09-14 at 9 34 33 AM" src="https://github.com/user-attachments/assets/a483b0a1-d6a7-4c9d-b4a9-4844c632876e">

- include report comment text and the sender display names + avatars in the email body
- are sent from “Expensify Chat” if the report is a policy room
- are sent from “Expensify Concierge” otherwise

More screenshots highlighting different scenarios found in [this Google doc](https://docs.google.com/document/d/1QUkuWPqxuGBxdQ2m_qLLHzkljXpIrzfAkfhMfE56VBA/edit?usp=sharing)

_**This leads to a pretty confusing experience that also implies a lack of polish for a couple of reasons:**_
* When we send emails for comments from a single user, we don’t include any information about the commenter (name/avatar) in the body of the email. This is SUPER confusing if the commenter is a non-employee because the message would show as sent from “Expensify Chat/Concierge” and there’s no way to tell who the commenter was.
* Emails for comments sent by multiple users include nicely styled HTML that mirrors the experience in the NewDot App, while emails for comments sent by a single user only have raw text and are inconsistent with the rest of our design.
* Some emails are sent from Expensify Chat, some are sent from Expensify Concierge, and others appear as sent from the report commenter for internal employees, seemingly without any rhyme or reason. I had to look at the code and do a bunch of manual testing to figure out what was going on.

### Solution:
Standardize all of our report comment notifications to be an improved version of the second scenario from above, i.e.
* The report commenter’s name and avatar are included with the report comment text in the body of the message. 
* The email shows as sent from “Expensify” if there are multiple commenters, and shows as from “(SenderName) via Expensify” if there is only one commenter.

**_This means that all the information needed to understand the report comment will be included in the email body, and that all emails will be consistently styled to match the NewDot experience._**

Some additional points for completeness/discussion:
* Messages in the Concierge DM will appear as from “Concierge via Expensify”.
* The email avatar should always be the Expensify logo, and we’ll include the report commenter’s avatar in the email body.
* Note that these changes will only involve report comments sent in chat reports. We’re not including other report types (IOU/Task/etc.) in the conversation, their behavior should be left untouched.

**Comment by jasperhuangg** [comment]
P/S statement discussion happening [here](https://expensify.slack.com/archives/C03U7DCU4/p1726468334104699)

**Comment by jasperhuangg** [comment]
More discussion happening on Slack. It looks like once that wraps up the next steps would be a pre-design to really iron out the details of the implementation.

**Comment by jasperhuangg** [comment]
Updated P/S statement outlined [here](https://expensify.slack.com/archives/C03U7DCU4/p1726719915761239?thread_ts=1726468334.104699&cid=C03U7DCU4), we're going a simpler route that will solve this problem without changing too much of the existing flow.

**Comment by jasperhuangg** [comment]
Plate full at the moment after being auto-assigned a couple of issues, should be able to get to this some time this week.

**Comment by jasperhuangg** [comment]
Posted a new point of discussion [here](https://expensify.slack.com/archives/C03U7DCU4/p1727210537411719?thread_ts=1726468334.104699&cid=C03U7DCU4)

**Comment by jasperhuangg** [comment]
Making good progress with a draft PR: https://github.com/Expensify/Web-Expensify/pull/43681

**Comment by jasperhuangg** [comment]
PR was deployed to production last week. I think we can close this out!
